docs/source/reference/kubernetes/index.rst

```diff
@@ -0,0 +1,154 @@
+
+0. Make sure `kubectl <https://kubernetes.io/docs/tasks/tools/>`_, ``socat`` and ``lsof`` are installed.
+
+1. Place your kubeconfig file at ``~/.kube/config``.
+
+   .. code-block:: console
+
+     $ mkdir -p ~/.kube
+     $ cp /path/to/kubeconfig ~/.kube/config
+
+   You can verify your credentials are setup correctly by running :code:`kubectl get pods`
+
+2. Run :code:`sky check` and verify that Kubernetes is enabled in SkyPilot.
+
+   .. code-block:: console
+
+     $ sky check
+
+     Checking credentials to enable clouds for SkyPilot.
+     ...
+     Kubernetes: enabled
+     ...
+
+
+   .. note::
+     :code:`sky check` will also check if GPU support is available on your cluster. If GPU support is not available, it
+     will show the reason.
+     To setup GPU support on the cluster, refer to the :ref:`Kubernetes cluster setup guide <kubernetes-setup>`.
+
+4. You can now run any SkyPilot task on your Kubernetes cluster.
+
+   .. code-block:: console
+
+        $ sky launch --cpus 2+
+        == Optimizer ==
+        Target: minimizing cost
+        Estimated cost: $0.0 / hour
+
+        Considered resources (1 node):
+        ---------------------------------------------------------------------------------------------------
+         CLOUD        INSTANCE          vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE   COST ($)   CHOSEN
+        ---------------------------------------------------------------------------------------------------
+         Kubernetes   2CPU--2GB         2       2         -              kubernetes    0.00          ✔
+         AWS          m6i.large         2       8         -              us-east-1     0.10
+         Azure        Standard_D2s_v5   2       8         -              eastus        0.10
+         GCP          n2-standard-2     2       8         -              us-central1   0.10
+         IBM          bx2-8x32          8       32        -              us-east       0.38
+         Lambda       gpu_1x_a10        30      200       A10:1          us-east-1     0.60
+        ---------------------------------------------------------------------------------------------------.
+
+
+.. note::
+  SkyPilot will use the cluster and namespace set in the ``current-context`` in the
+  kubeconfig file. To manage your ``current-context``:
+
+  .. code-block:: console
+
+    $ # See current context
+    $ kubectl config current-context
+
+    $ # Switch current-context
+    $ kubectl config use-context mycontext
+
+    $ # Set a specific namespace to be used in the current-context
+    $ kubectl config set-context --current --namespace=mynamespace
+
+FAQs
+----
+
+* **Are autoscaling Kubernetes clusters supported?**
+
+  Yes - however they currently require adjusting the resource provisioning timeout (:code:`Kubernetes.TIMEOUT` in `clouds/kubernetes.py`) to a large value to give enough time for the cluster to autoscale. We are working on a better interface to adjust this timeout - stay tuned!
```

Comment:
"Yes" -> "Partially. To use it, they require ..."?

Reason: if the default doesn't work for common autoscaling setup then it sounds more like partially?

Not feeling strongly about this one.

Reply:
Reworded a bit, wdyt?